client/app/components/composites/MenuMobile/OffScreenMenu.js

```diff
@@ -5,21 +5,26 @@ import css from './MenuMobile.css';
 import MenuSection from './MenuSection';
 import Avatar from '../../elements/Avatar/Avatar';
 import AddNewListingButton from '../../elements/AddNewListingButton/AddNewListingButton';
+import LoginLinks from '../../composites/LoginLinks/LoginLinks';
 
 class OffScreenMenu extends Component {
 
   render() {
     const isOpenClass = this.props.isOpen ? css.offScreenMenuOpen : '';
 
+    const header = this.props.avatar ? [
+      this.props.avatar ? r(Avatar, this.props.avatar) : null,
```

Comment:
This check is probably redundant, if it's also checked before entering the branch. I think detecting login state by presence of `avatar` might not work though, I'd also do the backend changes in this branch or soon after merge.